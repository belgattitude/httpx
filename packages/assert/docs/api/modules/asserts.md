[@httpx/assert - v0.0.1](../README.md) / asserts

# Module: asserts

## Table of contents

### Functions

- [assertPlainObject](asserts.md#assertplainobject)
- [assertStrNotEmpty](asserts.md#assertstrnotempty)
- [assertUuidV4](asserts.md#assertuuidv4)

## Functions

### assertPlainObject

▸ **assertPlainObject**(`v`, `msgOrErrorFactory?`): asserts v is string

Assert string is not empty (trims the string by default)

#### Parameters

| Name | Type |
| :------ | :------ |
| `v` | `unknown` |
| `msgOrErrorFactory?` | `MsgOrErrorFactory` |

#### Returns

asserts v is string

___

### assertStrNotEmpty

▸ **assertStrNotEmpty**(`v`, `msgOrErrorFactory?`, `trim?`): asserts v is string

Assert string is not empty (trims the string by default)

#### Parameters

| Name | Type | Description |
| :------ | :------ | :------ |
| `v` | `unknown` | - |
| `msgOrErrorFactory?` | `MsgOrErrorFactory` | - |
| `trim?` | `boolean` | auto-trim, default true |

#### Returns

asserts v is string

___

### assertUuidV4

▸ **assertUuidV4**(`v`, `msgOrErrorFactory?`): asserts v is string

#### Parameters

| Name | Type |
| :------ | :------ |
| `v` | `unknown` |
| `msgOrErrorFactory?` | `MsgOrErrorFactory` |

#### Returns

asserts v is string
