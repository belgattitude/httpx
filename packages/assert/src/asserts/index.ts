export * from './object.asserts';
export * from './string.asserts';
export * from './uuid.asserts';
