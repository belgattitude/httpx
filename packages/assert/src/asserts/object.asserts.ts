import { isPlainObject } from '../guards';
import type { MsgOrErrorFactory } from '../types/internal.types';
import { createAssertException } from '../utils/createAssertException';

/**
 * Assert string is not empty (trims the string by default)
 */
export function assertPlainObject(
  v: unknown,
  msgOrErrorFactory?: MsgOrErrorFactory
): asserts v is string {
  if (!isPlainObject(v)) {
    throw createAssertException(msgOrErrorFactory);
  }
}
