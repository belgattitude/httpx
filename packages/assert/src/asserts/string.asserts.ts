import { isStrNotEmpty } from '../guards';
import type { MsgOrErrorFactory } from '../types/internal.types';
import { createAssertException } from '../utils/createAssertException';

/**
 * Assert string is not empty (trims the string by default)
 */
export function assertStrNotEmpty(
  v: unknown,
  msgOrErrorFactory?: MsgOrErrorFactory,
  /** auto-trim, default true */
  trim?: boolean
): asserts v is string {
  if (!isStrNotEmpty(v, trim ?? true)) {
    throw createAssertException(msgOrErrorFactory);
  }
}
