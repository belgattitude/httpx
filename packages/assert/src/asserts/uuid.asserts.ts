import { isUuidV4 } from '../guards';
import type { UuidV4 } from '../types';
import type { MsgOrErrorFactory } from '../types/internal.types';
import { createAssertException } from '../utils/createAssertException';

export function assertUuidV4(
  v: unknown,
  msgOrErrorFactory?: MsgOrErrorFactory
): asserts v is UuidV4 {
  if (!isUuidV4(v)) {
    throw createAssertException(msgOrErrorFactory, 'Value is not a uuid v4');
  }
}
