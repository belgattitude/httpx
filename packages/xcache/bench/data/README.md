Examples taken from https://microsoftedge.github.io/Demos/json-dummy-data/5MB.json