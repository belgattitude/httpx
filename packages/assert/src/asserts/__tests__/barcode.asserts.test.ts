import { eansTestData } from '../../../test/test.data';
import { assertEan13 } from '../barcode.asserts';

describe('barcode assertions tests', () => {
  it('should not throw when barcode is valid', () => {
    expect(() => assertEan13(eansTestData.ean13)).not.toThrow();
  });
  it('should throw when barcode is invalid', () => {
    expect(() => assertEan13('12345')).toThrowErrorMatchingSnapshot();
  });
  it('should throw custom error when barcode is invalid', () => {
    const e = new Error('cool');
    expect(() => assertEan13('12345', () => e)).toThrowErrorMatchingSnapshot();
  });
});
