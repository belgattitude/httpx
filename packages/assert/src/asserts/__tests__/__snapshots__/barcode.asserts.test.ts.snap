// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`barcode assertions tests > should throw custom error when barcode is invalid 1`] = `[Error: cool]`;

exports[`barcode assertions tests > should throw when barcode is invalid 1`] = `[Error: Invalid ean13 barcode]`;
